simulator/func_sim/func_sim.h

```diff
@@ -50,7 +50,8 @@ class FuncSim : public BasicFuncSim
         void write_register( Register index, uint64 value) { rf.write( index, narrow_cast<RegisterUInt>( value)); }
 
     public:
-        FuncSim( Endian endian, bool log, std::string_view isa = "");
+        FuncSim( Endian endian, bool log);
+        FuncSim( Endian endian, bool log, std::string_view isa);
```

Comment:
https://github.com/MIPT-ILab/mipt-mips/blob/686f672a61ce77c6394bdfdf53da71c3bb5fea5c/simulator/export/cen64/cen64_wrapper.h#L29

https://github.com/MIPT-ILab/mipt-mips/blob/686f672a61ce77c6394bdfdf53da71c3bb5fea5c/simulator/modules/writeback/checker/checker.cpp#L16

There simulators are initialized without ISA.

Reply:
If you do not understand something, it's better to ask than to ignore.

For the first one, ISA is "mips64be" as VR4300 is a MIPS64 BE CPU.

The second one is a checker simulator. It is a functional simulator inside a performance simulator which executes the same program to check that performance simulation has no bugs. You should initialize it with the same ISA as performance simulator.

I believe here is the root cause of the issues you see with performance modeling. Checker has the kernel memory image (as memory is duplicated), but it does not have CSR values. I need to think how it can be fixed properly.